src/main/java/spoon/reflect/visitor/filter/Scann.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (C) 2006-2016 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.visitor.filter;
+
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.visitor.CtScanner;
+import spoon.reflect.visitor.chain.AsyncFunction;
+import spoon.reflect.visitor.chain.Consumer;
+
+public class Scann extends CtScanner implements AsyncFunction<CtElement, CtElement> {

```

Comment:
It can be used in the QueryStep chain to visit all the children of chain input element.
It is used automatically In combination with filter in method `QuertStep#scan(Filter)`, but I guess it can be use independent using ...query().then(new Scann())....